### ISSUE TITLE
SPV mode for pynode

### ISSUE THREAD

**Issue by amiller** [jgarzik/pynode#15]
pynode runs mostly like a full-validating node, meaning it downloads all of every block, maintains an entire UTXO set, etc.

Especially now that most of pynode is factored out, we might want to have a lite version of pynode, similar to BitcoinJ. This would also be a first step towards light wallet and contract apps.

A lite node should have much less functionality, for example it need not necessarily serve transactions and blocks to other peers, it shouldn't store archival history of blocks or the whole ledger, validate blocks, or even download all the transaction in each block.

This attempt is beginning in, and features a new database file LiteDb, and a new driver/event loop spvnode.py:
https://github.com/amiller/pynode/tree/spv

**Comment by jgarzik** [comment]
Agreed.  I'm happy to merge changes like this into upstream pynode.
